### ISSUE TITLE
Is the centerness branch able to replace as a centerness loss in target function and only be used in inference process?

### ISSUE THREAD

**Issue by foreverYoungGitHub** [tianzhi0549/FCOS#270]
Hi, @tianzhi0549,

Thank you for your great work!

I'm really curious about the centerness branch in the FCOS. It has been proved by FCOS and ATSS paper that it can improve the percision of model significantly, espectially in the reg branch.

However, based on eq. 3) in the paper. The centerness target is directly calculated by the regression target. Is that possible we could replace the centerness branch as a penalty in the regression loss in the training process? 

Also since it is directly calculated by the reg target, in the inference process, the centerness actually can be directly calculated by the reg prediction and applied it to the score value to filter the low quality box as well.

I'm wondering what is the advantage to adopt centerness branch in FCOS and why does it have this effect?

Thanks!

Best,

**Comment by Latootoo** [comment]
Centerness branch is introduced to align the cls_score and bbox_pred.
In the training step, the target centerness can be used to down-weight IoU loss of predicted boxes near the edge. 
In the inference step, the predicted centerness is used to down-weight the  cls-scores of predicted boxes near the edge.
